parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -613,7 +613,28 @@ installKubeletKubectlAndKubeProxy() {
             if [[ ! -z ${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER} ]]; then
                 # network isolated cluster
                 echo "Detect Bootstrap profile artifact is Cache, will use oras to pull artifact binary"
-                registry_url="${BOOTSTRAP_PROFILE_CONTAINER_REGISTRY_SERVER}/${K8S_REGISTRY_REPO}/kubernetes-node:v${KUBERNETES_VERSION}-linux-${CPU_ARCH}"
+                binary_version="v${KUBERNETES_VERSION}" # by default use Kubernetes versions
+                # if rp already passes registry url, then directly use the registry url that rp passes
+                # this path should have not catch for now, but keep it for future 
+                if [[ -n "${KUBE_BINARY_URL}" ]] && isRegistryUrl "${KUBE_BINARY_URL}"; then
+                    echo "KUBE_BINARY_URL is a registry url, will use it to pull the kube binary"
+                    registry_url="${KUBE_BINARY_URL}"
+                else
+                    # however, the kubelet and kubectl binary version may different with kubernetes version due to hotfix or beta
+                    # so that we still need to extract the binary version from kube_binary_url
+                    url_regex='https://[^/]+/kubernetes/v[0-9]+\.[0-9]+\..+/binaries/.+'
```

Comment:
if you would, please add some new unit tests to cover these different cases: https://github.com/Azure/AgentBaker/blob/dev/spec/parts/linux/cloud-init/artifacts/cse_install_spec.sh

Reply:
otherwise lgtm